// croxy-api/app.js
const request = require('node-superfetch');
const deasync = require('deasync');

class CroxyApi {
    constructor(keys) {
        this.key = keys;
        if (!this.key) throw new Error('Lütfen bir key giriniz. (Key almak için: https://croxy.codes/ sitesinden ya da Discord sunucumuza gelerek alabilirsiniz: https://discord.gg/mswG4px)');
        new Promise(async function(resolve, reject){
            const { body } = await request.get('https://croxy-api.glitch.me/keychecker/'+keys);
            if(body.key === false) throw new Error('Girdiğiniz key geçersizdir.')
            global.key = keys
        })
    };

    static async döviz() {
        try {
            const { body } = await request.get('https://croxy-api.glitch.me/doviz/'+global.key);
            let veriler = {};
            body.forEach(v => {
                veriler[v.birim] = {
                "birim": v.birim,
                "isim": v.isim,
                "alış": v.alış,
                "satış": v.satış
                }
            });
            return veriler
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async deprem(sayi) {
        if(!sayi || !Number(sayi) || sayi < 1) throw new TypeError('Lütfen kaç tane depremin görüntüleceğini yazın.')
        try {
            const { body } = await request.get('https://croxy-api.glitch.me/deprem/'+sayi+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async mcsunucu(yazi) {
    if(!yazi) throw new TypeError('Lütfen bir sunucu adresi gir. Örneğin; mc.craftrise.tc')
        try {
            const { body } = await request.get("https://croxy-api.glitch.me/minecraft/sunucu/"+yazi.toLowerCase()+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async mcoyuncu(yazi) {
        if(!yazi) throw new TypeError('Lütfen bir oyuncu ismi gir. Örneğin; BurakGider')
        try {
          const { body } = await request.get("https://croxy-api.glitch.me/minecraft/oyuncu/"+yazi.toLowerCase()+'/'+global.key);
          return body
      } catch (err) {
        if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
        throw new TypeError(err);
      }
    }

    static async twitter(yazi) {
        if(!yazi) throw new TypeError('Lütfen bir hesap ismi gir. Örneğin; bburakgiderr')
        try {
            const { body } = await request.get("https://croxy-api.glitch.me/twitter/"+yazi+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }

    static async film(yazi) {
        if(!yazi) throw new TypeError('Lütfen bir film ismi gir. Örneğin; Ayla')
        try {
            const { body } = await request.get("https://croxy-api.glitch.me/film/"+yazi+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }

    static async dizi(yazi) {
        if(!yazi) throw new TypeError('Lütfen bir dizi ismi gir. Örneğin; Stranger Things')
        try {
            const { body } = await request.get("https://croxy-api.glitch.me/dizi/"+yazi+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async şarkı(yazi) {
        if(!yazi) throw new TypeError('Lütfen bir şarkı ismi gir. Örneğin; Ezhel Felaket')
        try {
            const { body } = await request.get("https://croxy-api.glitch.me/sarki/"+yazi.toLowerCase()+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async şifre(sayi) {
        if(!sayi) throw new TypeError('Lütfen şifrenizin uzunluğunu belirleyin')
        if(isNaN(sayi)) throw new TypeError('Lütfen bir sayı gir.')
        try {
            const { body } = await request.get("https://croxy-api.glitch.me/sifreolustur/"+sayi+'/'+global.key);
            return body.hata ? body.hata : body.sonuç
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async birimçevir(birim, sayi) {
        const birimler = ["USD", "AUD", "DKK", "EUR", "GBP", "CHF", "SEK", "CAD", "KWD", "NOK", "SAR", "JPY", "BGN", "RON", "RUB", "IRR", "CNY", "PKR", "QAR"]
        if(!birim) throw new TypeError('Lütfen bir birim giriniz.')
        if(!birimler.includes(birim)) throw new TypeError('Lütfen geçerli bir birim giriniz. Bütün birimleri öğrenmek için https://croapi.cf/docs/birimler')
        if(!sayi) throw new TypeError('Lütfen bir sayı girin.')
        if(isNaN(sayi)) throw new TypeError('Lütfen bir sayı girin.')
        try {
            const { body } = await request.get(`https://croxy-api.glitch.me/birimcevir/${birim}/${sayi}`+'/'+global.key);
            return body
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async çeviri(dil, yazi) {
        if(!dil) throw new TypeError('Lütfen çevirilecek dili girin.')
        if(!yazi) throw new TypeError('Lütfen çevireceğiniz yazıyı girin.')
        try {
            const { body } = await request.get(`https://croxy-api.glitch.me/ceviri/${dil}/${yazi}`+'/'+global.key);
            if(body.sonuç === yazi) throw new TypeError("Lütfen geçerli bir dil girin. Örneğin; tr, es, en vb.")
            return body.hata ? body.hata : body.sonuç
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async tarih(date, format) {
        if(!date) throw new TypeError('Lütfen bir tarih girin. Örneğin; 81204827 ya da January 1, 2019')
        try {
            if(!format) {
                var { body } = await request.get(`https://croxy-api.glitch.me/tarih/normal/${date}`+'/'+global.key);
                return body.hata ? body.hata : body.sonuç
            } else {
                var { body } = await request.get(`https://croxy-api.glitch.me/tarih/normal/${date}/${format}`+'/'+global.key);
                return body.hata ? body.hata : body.sonuç
            }
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }

    static async rastgele(val) {
        try {
            var { body } = await request.get(`https://croxy-api.glitch.me/rastgeleveri/${val}`+'/'+global.key);
            return body.hata ? body.hata : body.sonuç
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
    static async kalansüre(val) {
        if(!val) throw new TypeError('Lütfen bir tarih girin. Örneğin; 101204827 ya da January 1, 2022')
        try {
            var { body } = await request.get(`https://croxy-api.glitch.me/kalansure/${val}`+'/'+global.key);
            return body.hata ? body.hata : body.sonuç
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
    static async geçensüre(val) {
        if(!val) throw new TypeError('Lütfen bir tarih girin. Örneğin; 81204827 ya da January 1, 2020')
        try {
            var { body } = await request.get(`https://croxy-api.glitch.me/gecensure/${val}`+'/'+global.key);
            return body.hata ? body.hata : body.sonuç
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }
      
    static async sayısalTarih(date, format) {
        if(!date) throw new TypeError('Lütfen bir tarih girin. Örneğin; 81204827 ya da January 1, 2019')
        try {
            if(!format) {
                var { body } = await request.get(`https://croxy-api.glitch.me/tarih/sayisal/${date}`+'/'+global.key);
                return body.hata ? body.hata : body.sonuç
            } else {
                var { body } = await request.get(`https://croxy-api.glitch.me/tarih/sayisal/${date}/${format}`+'/'+global.key);
                return body.hata ? body.hata : body.sonuç
            }
        } catch (err) {
            if(err.message === "503 Service Unavailable") throw new TypeError("Verilerin geldiği sitede bir hata ortaya çıktı. Lütfen tekrar deneyin. Eğer hala çalışmazsa lütfen modülün Discord destek sunucusuna gelin. https://discord.gg/mswG4px")
            throw new TypeError(err);
        }
    }

};
module.exports = CroxyApi;
module.exports.version = require('./package.json').version
var functions = {
   şarkı: function(x) { return CroxyApi.şarkı(x) },
   döviz: function(x) { return CroxyApi.döviz(x) },
   deprem: function(x) { return CroxyApi.deprem(x) },
   mcsunucu: function(x) { return CroxyApi.mcsunucu(x) },
   mcoyuncu: function(x) { return CroxyApi.mcoyuncu(x) },
   twitter: function(x) { return CroxyApi.twitter(x) },
   film: function(x) { return CroxyApi.film(x) },
   dizi: function(x) { return CroxyApi.dizi(x) },
   şifre: function(x) { return CroxyApi.şifre(x) },
   birimçevir: function(x, y) { return CroxyApi.birimçevir(x, y) },
   çeviri: function(x, y) { return CroxyApi.çeviri(x, y) },
   tarih: function(x, y) { return CroxyApi.tarih(x, y) },
   sayısalTarih: function(x, y) { return CroxyApi.sayısalTarih(x, y) },
   rastgele: function(x) { return CroxyApi.rastgele(x) },
   geçensüre: function(x) { return CroxyApi.geçensüre(x) },
   kalansüre: function(x) { return CroxyApi.kalansüre(x) },
}

module.exports.functions = functions;
